### ISSUE TITLE
Return a non-zero exit code (or a flag for that option) when files fail to update on `hs upload` command

### ISSUE THREAD

**Issue by stephenandersondev** [HubSpot/hubspot-cli#626]
## Description and Context

Could a non-zero exit code be returned when a file doesn't upload on the `hs upload` command due to an error? We run this command with a GitHub deploy action and saw that the deployment was showing to be successful even though some module files didn't update due to errors.

## Expected behavior

I would expect to receive a non-zero exit code to alert that there was an error with the upload, as of right now you won't know until you notice some of your modules didn't update.

## Screenshots

![Screen Shot 2022-01-07 at 10 03 05 AM](https://user-images.githubusercontent.com/65314061/148572546-7a2bea23-1fcd-43ea-9085-029dbf3d29b6.png)

**Comment by wreiske** [comment]
https://github.com/HubSpot/hubspot-cli/blob/master/packages/cli/commands/upload.js#L57-L59

https://github.com/HubSpot/hubspot-cli/blob/7667862d4e75e8f896b25030956dcfc1dc6124ca/packages/cli/README.md#exit-codes

I haven't dug too deep into this... It looks like it should be exiting with the error modes, but possibly only when a specific validation mode is set.

It's also possible that how we run it `npx hs upload` is not getting passed the exit code from upload 🤔

**Comment by SteffenMahler** [comment]
Ok, I am **not** a javascript developer, but I tried and failed in the first step.
**Simply adding a process.exit did not work.**

To reproduce the error case, I did the following:
```
hs create webpack-serverless
```
This command created a folder with the webpack-serverless boilerplate code.

Then I tried to upload this folder with
```
hs upload webpack-serverless webpack-serverless
```
This fails with
```
[ERROR] Uploading file "..."
[ERROR] Failed to read the source for the serverless config file
```


**Second attempt**
I took a closer look ath the "uploadFolder" method in uploadFolder.js and came across another catch(error) block in line 87 with error handling and logging. The problem is that the error is not passed on to the parent method in all cases.

After a little fix I got this :-)

![image](https://user-images.githubusercontent.com/57138594/154794316-3597744a-ac8f-4f8f-ae95-f9048081c8cb.png)
